client.go

```diff
@@ -161,8 +164,17 @@ func (c *Client) Do(ctx context.Context, req *http.Request) (resp *http.Response
 	}
 
 	// If 401, get new token and retry one time.
+	tcr := &TokenCreateRequest{}
 	if resp.StatusCode == http.StatusUnauthorized {
-		tok, tokErr := c.Token.Create(ctx, &TokenCreateRequest{Username: c.username, Password: c.password})
+		switch c.authMethod {
```

Comment:
Don't do this here, do it in Token.Create instead

Reply:
Then we need to pass `c *Client` to `Token.Create` function ?

Reply:
There has been `t.client` already.